src/main/java/graphql/GraphQL.java

```diff
@@ -289,18 +300,27 @@ public ExecutionResult execute(ExecutionInput executionInput) {
     }
 
     private ExecutionResult parseValidateAndExecute(ExecutionInput executionInput) {
-        ParseResult parseResult = parse(executionInput);
-        if (parseResult.isFailure()) {
-            return toParseFailureExecutionResult(parseResult.getException());
+        PreparsedDocumentEntry preparsedDoc = preparsedDocumentProvider.get(executionInput.getQuery(), query -> {
```

Comment:
Converted lambda to function, will look into if I can target this function with test.